crates/subspace-farmer/src/farm/plotted_pieces.rs

```diff
@@ -61,7 +61,7 @@ where
     pub fn read_piece(
         &self,
         piece_index: PieceIndex,
-    ) -> Option<impl Future<Output = Option<Piece>> + 'static> {
+    ) -> Option<impl Future<Output = Option<Piece>> + use<FarmIndex>> {
```

Comment:
As far as I can see, the `use<FarmIndex>` is only required for debugging, so it can be removed by doing:
```rust
let farm_index = usize::from(piece_details.farm_index);
```
near the start of the function.

This doesn't need to be in this PR.